client/src/connection/itc/index.ts

```diff
@@ -0,0 +1,413 @@
+  Write-Host $log
+} while ($log.Length -gt 0)\n
+  `,
+      this.onWriteComplete,
+    );
+  };
+
+  /**
+   * Handles stderr output from the powershell child process.
+   * @param chunk a buffer of stderr output from the child process.
+   */
+  private onShellStdErr = (chunk: Buffer): void => {
+    const msg = chunk.toString();
+    console.warn("shellProcess stderr: " + msg);
+    this.clearPassword();
+    this._runReject(
+      new Error(
+        "There was an error executing the SAS Program.\nSee console log for more details.",
+      ),
+    );
+    // If we encountered an error in setup, we need to go through everything again
+    if (/Setup error/.test(msg)) {
+      this._shellProcess.kill();
+      this._workDirectory = undefined;
+    }
+  };
+
+  /**
+   * Handles stdout output from the powershell child process.
+   * @param data a buffer of stdout output from the child process.
+   */
+  private onShellStdOut = (data: Buffer): void => {
+    const output = data.toString().trimEnd();
+    const outputLines = output.split(/\n|\r\n/);
+
+    outputLines.forEach((line: string) => {
+      if (!line) {
+        return;
+      }
+      if (!this._workDirectory && line.startsWith("WORKDIR=")) {
+        const parts = line.split("WORKDIR=");
+        this._workDirectory = parts[1].trim();
+        this._runResolve();
+        return;
+      }
+      if (!this.processLineCodes(line)) {
+        this._html5FileName =
+          line.match(/NOTE: .+ HTML5.* Body.+: (.+)\.htm/)?.[1] ??
+          this._html5FileName;
+        this._onLogFn?.([{ type: "normal", line }]);
+      }
+    });
+  };
+
+  private processLineCodes(line: string): boolean {
+    if (line.includes(LineCodes.RunEndCode)) {
+      // run completed
+      this.fetchResults();
+      return true;
+    }
+
+    if (line.includes(LineCodes.SessionCreatedCode)) {
+      this.storePassword();
+      return true;
+    }
+
+    if (line.includes(LineCodes.ResultsFetchedCode)) {
+      this.displayResults();
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Generic call for use on stdin write completion.
+   * @param err The error encountered on the write attempt. Undefined if no error occurred.
+   */
+  private onWriteComplete = (err: Error): void => {
+    if (err) {
+      this._runReject?.(err);
+    }
+  };
+
+  /**
+   * Not implemented.
+   */
+  public sessionId = (): string => {
+    throw new Error("Not Implemented");
+  };
+
+  /**
+   * Fetches the ODS output results for the latest html results file.
+   */
+  private fetchResults = async () => {
+    const globalStorageUri = getGlobalStorageUri();
+    try {
+      await workspace.fs.readDirectory(globalStorageUri);
+    } catch (e) {
+      await workspace.fs.createDirectory(globalStorageUri);
+    }
+
+    const outputFileUri = Uri.joinPath(
+      globalStorageUri,
+      `${this._html5FileName}.htm`,
+    );
+    this._shellProcess.stdin.write(
+      `$filePath = "${resolve(
```

Comment:
The `resolve` will try to normalize the path based on the OS it is running on, which is Windows.
But the working directory is on the server that SAS is running on, which could be Linux.
The resulting directory may be unexpected. I believe that is the cause of the error Zhirong encountered.

Reply:
You're right. I've addressed this in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/61aab6cf6d682732700612cca59b520afae81d0e . Admittedly, my approach seems a _little_ clunky, but should support both COM/IOM Bridge connection types. 

@Zhirong2022 One thing to mention here is that we'll likely want to test COM connections as part of this